ci-operator/step-registry/ipi/conf/capability/additonalcaps/ipi-conf-capability-additonalcaps-commands.sh

```diff
@@ -0,0 +1,76 @@
+#!/bin/bash
+
+set -o nounset
+set -o errexit
+set -o pipefail
+
+if [[ "${BASELINE_CAPABILITY_SET}" == "" ]]; then
+    echo "This step requires BASELINE_CAPABILITY_SET to be set!"
+    exit 1
+fi
+
+if [[ "${ADDITIONAL_ENABLED_CAPABILITIES}" != "" ]]; then
+   echo "ENV 'ADDITIONAL_ENABLED_CAPABILITIES' is set, this step is not required!"
+   exit 0
+fi
+
+echo "RELEASE_IMAGE_LATEST: ${RELEASE_IMAGE_LATEST}"
+echo "RELEASE_IMAGE_LATEST_FROM_BUILD_FARM: ${RELEASE_IMAGE_LATEST_FROM_BUILD_FARM}"
+export HOME="${HOME:-/tmp/home}"
+export XDG_RUNTIME_DIR="${HOME}/run"
+export REGISTRY_AUTH_PREFERENCE=podman # TODO: remove later, used for migrating oc from docker to podman
+mkdir -p "${XDG_RUNTIME_DIR}"
+# After cluster is set up, ci-operator make KUBECONFIG pointing to the installed cluster,
+# to make "oc registry login" interact with the build farm, set KUBECONFIG to empty,
+# so that the credentials of the build farm registry can be saved in docker client config file.
+# A direct connection is required while communicating with build-farm, instead of through proxy
+KUBECONFIG="" oc --loglevel=8 registry login
+ocp_version=$(oc adm release info ${RELEASE_IMAGE_LATEST_FROM_BUILD_FARM} --output=json | jq -r '.metadata.version' | cut -d. -f 1,2)
+echo "OCP Version: $ocp_version"
+ocp_major_version=$( echo "${ocp_version}" | awk --field-separator=. '{print $1}' )
+ocp_minor_version=$( echo "${ocp_version}" | awk --field-separator=. '{print $2}' )
+
+v411="baremetal marketplace openshift-samples"
+v412=" ${v411} Console Insights Storage CSISnapshot"
+v413=" ${v412} NodeTuning"
+# shellcheck disable=SC2034
+v414=" ${v413} MachineAPI Build DeploymentConfig ImageRegistry"
+
+# shellcheck disable=SC2034
+v_current_version="v${ocp_major_version}${ocp_minor_version}"
+vcurrent_capabilities="${!v_current_version}"
+
+#Randomly select one capability to be disabled
+# shellcheck disable=SC2206
+vcurrent_capabilities_array=(${vcurrent_capabilities})
+selected_capability_index=$((RANDOM % ${#vcurrent_capabilities_array[@]}))
+selected_capability="${vcurrent_capabilities_array[$selected_capability_index]}"
+echo "Selected capability to be disabled: ${selected_capability}"
+
+enabled_capabilities=${vcurrent_capabilities}
+if [[ ! "${selected_capability}" == "MachineAPI" ]]; then
+    enabled_capabilities=${enabled_capabilities/${selected_capability}}
+    if [[ "${selected_capability}" == "ImageRegistry" ]]; then
```

Comment:
Sounds we move this ImageRegistry out of the MachineAPI if block to make them on the same level, maybe more easiy-reading?